### PR TITLE
Frontend: Base home/login, cars, events, profile pages

### PR DESCRIPTION
**Pull request by christerresin**
Based on Erm0es design and boilerplate for frontend. 

Additions/changes:
- Not so secure login
- Not so secure logic to check if logged in
- Boilerplate Events page
- Base Cars page with logic to fetch cars data from API
- Boilerplate Profile/Person page
- Placeholder image
- Refactoring of folder structure
- Helper functions for jwt checking/setting and logout
- React Router for navigation
- Axios for making API req

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -6,14 +6,16 @@
...
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-router-dom": "^6.4.5",
     "react-scripts": "5.0.1",
     "web-vitals": "^2.1.4"
   },
   "scripts": {
     "start": "react-scripts start",
-    "build": "react-scripts build",
+    "build": "CI=false react-scripts build",
```
Comment (stratosfloor) [inline frontend/package.json:18]: DU
ÄR
ETT
GENI!

File: backend/src/main/java/com/example/parkingspot/SecurityConfig.java [hunk 2]
```diff
@@ -12,43 +12,45 @@
...
-                  "/api/events/**").authenticated()
-                .requestMatchers(HttpMethod.PUT,
-                  "/api/cars/**",
-                  "/api/events/**").authenticated()
-                .anyRequest().denyAll()
-              .and()
-              .oauth2ResourceServer(OAuth2ResourceServerConfigurer::jwt)
-              .build();
+        .csrf().disable()
+        // .cors().disable()
```
Comment (amyrman) [inline backend/src/main/java/com/example/parkingspot/SecurityConfig.java]: Is enabling cors intentional?

File: backend/src/main/java/com/example/parkingspot/controller/PersonController.java [hunk 3]
```diff
@@ -17,6 +18,7 @@
 import com.example.parkingspot.service.CarService;
 import com.example.parkingspot.service.PersonService;
 
+@CrossOrigin
```
Comment (amyrman) [inline backend/src/main/java/com/example/parkingspot/controller/PersonController.java]: If we disable cors in SecurityConfig, will this be needed?

File: backend/src/main/java/com/example/parkingspot/SecurityConfig.java (hunk 2 above)
Comment (christerresin) [reply backend/src/main/java/com/example/parkingspot/SecurityConfig.java]: Great catch! Working on new push to again disable cors

File: backend/src/main/java/com/example/parkingspot/controller/PersonController.java (hunk 3 above)
Comment (christerresin) [reply backend/src/main/java/com/example/parkingspot/controller/PersonController.java]: Correct! This annotation does nothing if cors is disabled

**Comment by christerresin** [comment]
Updated - disabled cors and removed @CrossOrigin annotation

File: frontend/src/checkJwtExpired.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { logoutUser } from './logoutUser';
+
+export const checkJwtExpired = () => {
+  const token_duration = localStorage.getItem('token_duration');
+  if (Date.now() > token_duration) {
+    console.log('TOKEN EXPIRED!');
+    logoutUser();
+  } else {
```
Comment (stratosfloor) [inline frontend/src/checkJwtExpired.js:8]: What happends when user is logged out? 
Pushed to login page? "Session expired"

To do if we got time.

File: frontend/src/component/login.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+function Login() {
+  return (
+    <div className='page'>
+      <Header />
+      <LoginForm />
+
+      <button onClick={handleGetAllPersons}>GET PERSONS</button>
+      <button onClick={checkJwtExpired}>VALID?!</button>
+    </div>
```
Comment (stratosfloor) [inline frontend/src/component/login.js:12]: Why are these buttons on the login page? Just dev stuff?

**Review by stratosfloor — COMMENTED** [review]
Frontend is working but not acessing to backend. Do we need to implement Anders login for the backend first?

File: frontend/src/component/login.js (hunk 5 above)
Comment (christerresin) [reply frontend/src/component/login.js:12]: Dev stuff :D

**Review by amyrman — APPROVED** [review]
Looks good so far.

**Review by stratosfloor — APPROVED** [review]
Login endpoint works correctly and returns userId. 
Set userId and JWT to localstorage at login och fetch cars correctly.

**Pull request merged**